src/protocols/secure_channel/CASESession.cpp

```diff
@@ -2163,65 +2312,57 @@ CHIP_ERROR CASESession::OnFailureStatusReport(Protocols::SecureChannel::GeneralS
-    constexpr uint8_t kInitiatorMRPParamsTag = 5;
-    constexpr uint8_t kResumptionIDTag       = 6;
-    constexpr uint8_t kResume1MICTag         = 7;
-
     TLVType containerType = kTLVType_Structure;
     ReturnErrorOnFailure(tlvReader.Next(containerType, AnonymousTag()));
     ReturnErrorOnFailure(tlvReader.EnterContainer(containerType));
 
-    ReturnErrorOnFailure(tlvReader.Next(ContextTag(kInitiatorRandomTag)));
-    ReturnErrorOnFailure(tlvReader.GetByteView(initiatorRandom));
-    VerifyOrReturnError(initiatorRandom.size() == kSigmaParamRandomNumberSize, CHIP_ERROR_INVALID_CASE_PARAMETER);
+    ReturnErrorOnFailure(tlvReader.Next(ContextTag(ToRaw(Sigma1Tags::kInitiatorRandom))));
+    ReturnErrorOnFailure(tlvReader.GetByteView(output.initiatorRandom));
+    VerifyOrReturnError(output.initiatorRandom.size() == kSigmaParamRandomNumberSize, CHIP_ERROR_INVALID_CASE_PARAMETER);
 
-    ReturnErrorOnFailure(tlvReader.Next(ContextTag(kInitiatorSessionIdTag)));
-    ReturnErrorOnFailure(tlvReader.Get(initiatorSessionId));
+    ReturnErrorOnFailure(tlvReader.Next(ContextTag(ToRaw(Sigma1Tags::kInitiatorSessionId))));
+    ReturnErrorOnFailure(tlvReader.Get(output.initiatorSessionId));
 
-    ReturnErrorOnFailure(tlvReader.Next(ContextTag(kDestinationIdTag)));
-    ReturnErrorOnFailure(tlvReader.GetByteView(destinationId));
-    VerifyOrReturnError(destinationId.size() == kSHA256_Hash_Length, CHIP_ERROR_INVALID_CASE_PARAMETER);
+    ReturnErrorOnFailure(tlvReader.Next(ContextTag(ToRaw(Sigma1Tags::kDestinationId))));
+    ReturnErrorOnFailure(tlvReader.GetByteView(output.destinationId));
+    VerifyOrReturnError(output.destinationId.size() == kSHA256_Hash_Length, CHIP_ERROR_INVALID_CASE_PARAMETER);
 
-    ReturnErrorOnFailure(tlvReader.Next(ContextTag(kInitiatorPubKeyTag)));
-    ReturnErrorOnFailure(tlvReader.GetByteView(initiatorEphPubKey));
-    VerifyOrReturnError(initiatorEphPubKey.size() == kP256_PublicKey_Length, CHIP_ERROR_INVALID_CASE_PARAMETER);
+    ReturnErrorOnFailure(tlvReader.Next(ContextTag(ToRaw(Sigma1Tags::kInitiatorPubKey))));
+    ReturnErrorOnFailure(tlvReader.GetByteView(output.initiatorEphPubKey));
+    VerifyOrReturnError(output.initiatorEphPubKey.size() == kP256_PublicKey_Length, CHIP_ERROR_INVALID_CASE_PARAMETER);
 
     // Optional members start here.
     CHIP_ERROR err = tlvReader.Next();
-    if (err == CHIP_NO_ERROR && tlvReader.GetTag() == ContextTag(kInitiatorMRPParamsTag))
+    if (err == CHIP_NO_ERROR && tlvReader.GetTag() == ContextTag(ToRaw(Sigma1Tags::kInitiatorMRPParams)))
     {
-        ReturnErrorOnFailure(DecodeMRPParametersIfPresent(TLV::ContextTag(kInitiatorMRPParamsTag), tlvReader));
-        mExchangeCtxt.Value()->GetSessionHandle()->AsUnauthenticatedSession()->SetRemoteSessionParameters(
-            GetRemoteSessionParameters());
+        ReturnErrorOnFailure(DecodeMRPParametersIfPresent(TLV::ContextTag(ToRaw(Sigma1Tags::kInitiatorMRPParams)), tlvReader));
```

Comment:
TLV namespace would be good to have as `using` (looks like already done). Note that on line 2340 there was `ContextTag` instead of `TLV::ContextTag` and it did not fail to compile.

Reply:
this comment is not longer valid due to below comment on replacing `ToRaw` with `AsTlvContextTag`

Reply:
Done for all modified functions